unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/NotificationMessagesHUD/ChatNotificationController.cs

```diff
@@ -68,7 +70,12 @@ private void HandleMessageAdded(ChatMessage message)
                     ? "nearby"
                     : message.recipient);
                 if (channel?.Muted ?? false) return;
-                
+
+                if(notificationEntries.Contains(message.messageId))
+                    return;
```

Comment:
i think you should not return, instead the entry should be updated with the newest data to get a consistent state in memory. For example the timestamp of the message could change which may affect the sorting. Do you think it is feasible or it would require a major refactor?

Reply:
ok then, can you add a comment then? like: `TODO: entries may have an inconsistent state. We should update the entry with new data`